build_tools.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
```

Comment:
I think that it makes sense to split up the list of APT packages above between build and runtime dependencies, and then the chain together the `git clone`, `cmake`, `make` calls, followed by the removal of build dependencies (so that they're all in one `RUN`).

It could also make sense to split out specific APT packages that you know will not be removed into a separate layer so that the layer with all of the build stuff is more readable.

An example that contains bits of all of this is here: https://github.com/docker-library/redis/blob/master/3.2/Dockerfile

- Permentant deps: https://github.com/docker-library/redis/blob/master/3.2/Dockerfile#L6-L9
- APT install, download, extract, compile, cleanup: https://github.com/docker-library/redis/blob/master/3.2/Dockerfile#L28-L61

I could also see it being beneficial to switch the `git clone` steps so that we download the associated tarball for the specific ref. It might allow us to omit `git` and its dependencies entirely.